crates/extensions/c8y_mapper_ext/src/config.rs

```diff
@@ -55,11 +50,11 @@ impl C8yMapperConfig {
     ) -> Result<C8yMapperConfig, TEdgeConfigError> {
         let config_dir: PathBuf = config_dir.as_ref().into();
 
-        let logs_path = tedge_config.query(LogPathSetting)?;
-        let device_id = tedge_config.query(DeviceIdSetting)?;
-        let device_type = tedge_config.query(DeviceTypeSetting)?;
-        let service_type = tedge_config.query(ServiceTypeSetting)?;
-        let c8y_host = tedge_config.query(C8yHttpSetting)?.into();
+        let logs_path = Utf8PathBuf::from(tedge_config.read_string(ReadableKey::LogsPath)?);
```

Comment:
There is no need to convert the camino path to a string and then to a path.

```suggestion
        let logs_path = tedge_config.logs.path.to_owned();
```